backEnd/routes/userCRUD.js

```diff
@@ -2,6 +2,64 @@ const express = require('express');
 const router = express.Router();
 const { PrismaClient } = require('../generated/prisma')
 const prisma = new PrismaClient;
+const bcrypt = require('bcrypt')
+
+
+router.post("/signup", async (req, res) => {
+  const { username, password, email, name, phoneNumber, address} = req.body;
+
+  if (!username || !password) {
+    return res.status(400).json({ error: "Username and password are required." });
+  }
+
+  const existingUser = await prisma.user.findUnique({
+    where: { username }
+  });
+
+  if (existingUser) {
+      return res.status(400).json({ error: "Username already taken." });
+  }
+
+  const bcrypt = require("bcrypt");
+  const passwordHash = await bcrypt.hash(password, 10);
```

Comment:
would recommend for security to use a salt

https://en.wikipedia.org/wiki/Salt_(cryptography)